src/Plugin/Filter/UrlEmbedFilter.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\url_embed\Plugin\Filter\UrlEmbedFilter.
+ */
+
+namespace Drupal\url_embed\Plugin\Filter;
+
+use Drupal\Core\Annotation\Translation;
+use Drupal\Component\Utility\Html;
+use Drupal\filter\Annotation\Filtere;
+use Drupal\filter\FilterProcessResult;
+use Drupal\filter\Plugin\FilterBase;
+use Embed\Embed;
+
+/**
+ * Provides a filter to display embedded URLs based on data attributes.
+ *
+ * @Filter(
+ *   id = "url_embed",
+ *   title = @Translation("Display embedded URLs"),
+ *   description = @Translation("Embeds URLs using data attributes: url, source, oembed provider and view-mode."),
+ *   type = Drupal\filter\Plugin\FilterInterface::TYPE_TRANSFORM_REVERSIBLE
+ * )
+ */
+class UrlEmbedFilter extends FilterBase {
+
+  /**
+   * Replace the contents of a DOMNode.
+   *
+   * @param \DOMNode $node
+   *   A DOMNode or DOMElement object.
+   * @param string $content
+   *   The text or HTML that will replace the contents of $node.
+   */
+  protected function setDomNodeContent(\DOMNode $node, $content) {
```

Comment:
Created a followup just so that this is not overlooked: https://www.drupal.org/node/2547105